libqtile/widget/wlan.py

```diff
@@ -42,13 +44,24 @@ def __init__(self, **config):
         self.add_defaults(Wlan.defaults)
 
     def poll(self):
-        interface = Wireless(self.interface)
         try:
-            stats = Iwstats(self.interface)
-            quality = stats.qual.quality
-            essid = interface.getEssid()
-            return "{} {}/70".format(essid, quality)
-        except IOError:
-            logging.getLogger('qtile').error('%s: Probably your wlan device '
-                    'is switched off or otherwise not present in your system.',
+            interface = Wireless(self.interface)
+            try:
+                stats = Iwstats(self.interface)
+                quality = stats.qual.quality
+                essid = interface.getEssid()
+                return "{} {}/70".format(essid,quality)
+            except IOError:
+                logging.getLogger('qtile').error('%s: Probably your wlan device '
+                'is switched off or otherwise not present in your system.',
+                self.__class__.__name__)
+        except:
```

Comment:
I think you should define functions that return quality and essid, rather that putting it in a bare try/except. That is, define a function, such as `_get_wifi_stats`, in the `try`/`except ImportError` blocks at the top of the file, then call that here.